### ISSUE TITLE
Router.replace crashes if previous URL is malformed

### ISSUE THREAD

**Issue by rap2hpoutre** [zeit/next.js#6638]
# Bug report

## Describe the bug

When I arrive to a Next website with a malformed URL (let's say http://localhost:3000/?a=%E7, which has an ISO encoded char), it crashes at first call to `Router.replace`. Demo below:

![next](https://user-images.githubusercontent.com/1575946/54262811-81a0db80-456f-11e9-9f15-e5a8e1e979fa.gif)

> Unhandled Rejection (URIError): URI malformed

## To Reproduce
1. Create a next app with a single `pages/index.js` file
```jsx
import Router from "next/router";
export default function Home() {
  return (
    <div>
      Welcome to next.js!
      <div onClick={() => { Router.replace("/?lol=2"); }}>Click me</div>
    </div>
  );
}
```
2. start app (`npm run dev`) and GOTO `http://localhost:3000/?a=%E7`.
3. Click on the `Click me` div.
4. It crashes and display: `Unhandled Rejection (URIError): URI malformed`. Same problem in production (without message in browser, only in console).

## Expected behavior
**It should not crash**: URL crafted on **other websites** should not crash the **actual next powered website**. Ex: we are referenced on wikipedia and some links are wrong, it should not crash _our_ app.

## System information
 - OS/Browser: all.
 - Version of Next.js: 8.0

**Comment by ijjk** [comment]
@rap2hpoutre I just tested this on the latest canary `8.0.4-canary.10` and it appears to be working ok. I'm going to close this issue, if you're still having trouble with this feel free to comment.

**Comment by rap2hpoutre** [comment]
> it appears to be working ok

Ok thanks, I just tried with this version and it appears to be working ok for me too. 

Still: 
 - did you fixed something about this? (is there a link to the relevant commit ?) Or is it a side effect caused by some other fix? (I tried to search about this issue and did not found anything...)
 - Where can I learn more about the release date plan? (in other words, when will this version be ready/stable)

Thanks!
